neural_modelling/src/neuron/c_main_neurons.c

```diff
@@ -271,6 +271,7 @@ static bool initialise(void) {
     spin1_memcpy(&sdram_inputs, sdram_config, sizeof(struct sdram_config));
 
     uint32_t n_words = sdram_inputs.size_in_bytes >> 2;
+    synaptic_contributions = spin1_malloc(N_SYNAPTIC_BUFFERS * sizeof(uint32_t*));
     for (uint32_t i = 0; i < N_SYNAPTIC_BUFFERS; i++) {
         synaptic_contributions[i] = spin1_malloc(sdram_inputs.size_in_bytes);
         if (synaptic_contributions == NULL) {
```

Comment:
The real error the warning is finding is here; it should be checking:
if (synaptic_contributions[i] == NULL) {